lib/puppet/type/hocon_setting.rb

```diff
@@ -44,7 +44,14 @@
             raise "Type specified as #{@resource[:type]} but was #{value.class}"
           end
         when 'number'
-          unless value.is_a?(Numeric)
+          # Puppet stringifies numerics in versions of Puppet < 4.0.0
+          # Account for this in the type
+          begin
+            numeric_as_string = Integer(value)
```

Comment:
@glarizza do we need to account for floats?  Would it cause any harm if we were to simply `s/Integer/Float/g` in your PR?